prime-router/src/main/kotlin/azure/ApiSearch.kt

```diff
@@ -0,0 +1,196 @@
+package gov.cdc.prime.router.azure
+
+import com.microsoft.azure.functions.HttpRequestMessage
+import gov.cdc.prime.router.common.JacksonMapperUtilities
+import org.jooq.Condition
+import org.jooq.DSLContext
+import org.jooq.Field
+import org.jooq.Record
+import org.jooq.SelectJoinStep
+import org.jooq.SortField
+import org.jooq.TableField
+
+data class ApiSearchResult<T>(val totalCount: Int, val filteredCount: Int, val results: List<T>)
+enum class SortDirection {
+    ASC,
+    DESC
+}
+
+/**
+ * Interface that can be extended to represent a set of search term names that can be applied
+ * to a specific search type
+ */
+interface ApiFilterNames
+
+/**
+ * Interface that represents an interface for retrieving a specific term
+ * @see ApiFilter
+ * @see gov.cdc.prime.router.db.ReportFileApiSearch.Companion
+ */
+interface ApiFilters<RecordType : Record, ApiFilterType : ApiFilter<RecordType, *>, Names : ApiFilterNames> {
+
+    val terms: Map<Names, Class<out ApiFilterType>>
+    fun getTerm(termName: Names): Class<out ApiFilterType>? {
+        return terms[termName]
+    }
+}
+
+interface ApiFilter<RecordType : Record, T> {
+
+    /** The value that will be applied to the operator */
+    val value: T
+
+    /** The field the value will be applied against */
+    val tableField: TableField<RecordType, T>
+}
+
+/**
+ * Abstract class that provides helpers for parsing either a query string or request body into a specific
+ * kind of ApiSearch
+ *
+ * @see gov.cdc.prime.router.db.ReportFileApiSearch.Companion
+ */
+abstract class ApiSearchParser<
+    PojoType,
+    ApiSearchType : ApiSearch<PojoType, RecordType, Names>,
+    RecordType : Record,
+    Names : ApiFilter<RecordType, *>
+    > {
+    /**
+     * Converts a query string into a RawApiSearch. Currently, not implemented
+     * TODO add ticket
```

Comment:
Probably needs to be removed.

Reply:
Good call!

Reply:
Added the ticket link to implement this.